op-dispute-mon/mon/node_endpoint_errors.go

```diff
@@ -0,0 +1,55 @@
+package mon
+
+import (
+	"github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type NodeEndpointErrorsMetrics interface {
+	RecordNodeEndpointErrors(count int)
+}
+
+type NodeEndpointErrorsMonitor struct {
+	logger  log.Logger
+	metrics NodeEndpointErrorsMetrics
+}
+
+func NewNodeEndpointErrorsMonitor(logger log.Logger, metrics NodeEndpointErrorsMetrics) *NodeEndpointErrorsMonitor {
+	return &NodeEndpointErrorsMonitor{
+		logger:  logger,
+		metrics: metrics,
+	}
+}
+
+func (m *NodeEndpointErrorsMonitor) CheckNodeEndpointErrors(games []*types.EnrichedGameData) {
+	// Use a set to track unique endpoint errors across all games
+	uniqueEndpointErrors := make(map[string]bool)
+
+	for _, game := range games {
+		if game.RollupEndpointErrors != nil {
+			for endpointID := range game.RollupEndpointErrors {
+				uniqueEndpointErrors[endpointID] = true
+			}
+		}
+	}
+
+	errorCount := len(uniqueEndpointErrors)
+	if errorCount > 0 {
+		m.logger.Warn("Found rollup node endpoint errors",
+			"unique_endpoint_count", errorCount,
+			"endpoints", getEndpointList(uniqueEndpointErrors))
+	} else {
+		m.logger.Debug("No rollup node endpoint errors found")
```

Comment:
nit: possibly noisy debug. Could use `Trace` instead.

Reply:
Removed it completely.